File: snippets/sequences/fibonacci.py
# Python integers are exact; no external libraries.
# Indexing: n starts at 0.
def nth_fibonacci(n):
    if not isinstance(n, int) or isinstance(n, bool) or not 0 <= n <= 1000:
        raise ValueError("n must be an integer from 0 to 1000")
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a

File: snippets/sequences/fibonacci.js
// Exact integers using built-in BigInt; no external libraries.
// Indexing: n starts at 0.
function nth_fibonacci(n) {
  if (!Number.isInteger(n) || n < 0 || n > 1000)
    throw new RangeError("n must be an integer from 0 to 1000");
  let a = 0n, b = 1n;
  for (let i = 0; i < n; i++) [a, b] = [b, a + b];
  return a;
}
